src/Controller/Admin/CategoryController.php

```diff
@@ -0,0 +1,148 @@
+                    'type' => 'text',
+                    'label' => 'common.name',
+                    'queryKey' => 'c.name',
+                ],
+                'updatedAt' => [
+                    'type' => 'date',
+                    'label' => 'common.updated_at',
+                    'queryKey' => 'c.updatedAt',
+                ],
+                'createdAt' => [
+                    'type' => 'date',
+                    'label' => 'common.created_at',
+                    'queryKey' => 'c.createdAt',
+                ],
+                'actions' => [
+                  'info' => [
+                      'route' => 'admin_category_show',
+                      'routeParams' => [
+                          'id' => static fn (Category $category): string => $category->getId()->toBase32(),
+                      ],
+                      'icon' => 'eye',
+                  ],
+                  'accent' => [
+                      'route' => 'admin_category_edit',
+                      'routeParams' => [
+                          'id' => static fn (Category $category): string => $category->getId()->toBase32(),
+                      ],
+                      'icon' => 'pencil',
+                  ],
+                ],
+            ],
+            'pagination' => $pagination,
+        ];
+
+        return $this->render('admin/category/index.html.twig', ['config' => $config]);
+    }
+
+    #[Route('/new', name: 'admin_category_new', methods: ['GET', 'POST'])]
+    public function new(Request $request): Response
+    {
+        return $this->form($request, new Category(), false);
+    }
+
+    #[Route('/{id}', name: 'admin_category_show', methods: ['GET'])]
+    public function show(Category $category): Response
+    {
+        return $this->render('admin/category/show.html.twig', [
+            'category' => $category,
+        ]);
+    }
+
+    #[Route('/{id}', name: 'admin_category_delete', methods: ['POST'])]
+    #[IsGranted(UserVoter::DELETE, subject: 'category', statusCode: 403)]
+    public function delete(Request $request, Category $category): Response
+    {
+        if ($this->isCsrfTokenValid('delete-'.$category->getId()->toBase32(), $request->request->get('_token'))) {
+            $this->categoryRepository->remove($category, true);
+
+            $this->flashManager->flash(ColorTypeEnum::Success->value, 'flash.common.deleted', translationDomain: 'admin');
+        } else {
+            $this->flashManager->flash(ColorTypeEnum::Error->value, 'flash.common.invalid_csrf');
+        }
+
+        return $this->redirectToRoute('admin_category', status: Response::HTTP_SEE_OTHER);
+    }
+
+    #[Route('/{id}/edit', name: 'admin_category_edit', methods: ['GET', 'POST'])]
+    public function edit(Request $request, Category $category): Response
+    {
+        return $this->form($request, $category, true);
+    }
+
+    public function form(Request $request, Category $category, bool $isEditing): Response
```

Comment:
C'est un peu bizarre de faire ça mais ok
Mettre en privé la fonction

```suggestion
    private function form(Request $request, Category $category, bool $isEditing): Response
```

Reply:
C'était pour éviter de dupliquer du code